src/controllers/LoginController.php

```diff
@@ -40,7 +40,14 @@ public function processLogin(): string {
                 $error = 'Username and password are required';
             } else {
                 if ($this->database->validateUser($username, $passwordHash)) {
-                    $error = "Login successful for $username";
+                    // Start session and store user information
+                    session_start();
+                    $_SESSION['user'] = $username;
+                    $_SESSION['logged_in'] = true;
```

Comment:
@jetbrains-junie store users id, instead of username. update / simplify related code accordingly.